airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/strip/pipeline.py

```diff
@@ -0,0 +1,195 @@
+      connector_dir_entries = await (await self.context.get_connector_dir()).entries()
+      
+      if self.context.connector.language != ConnectorLanguage.LOW_CODE:
+        return StepResult(
+          step=self,
+          status=StepStatus.SKIPPED,
+          stderr=f"The connector is not a low-code connector.",
+        )
+      
+      if self.context.connector.language == ConnectorLanguage.MANIFEST_ONLY:
+        return StepResult(
+          step=self,
+          status=StepStatus.SKIPPED,
+          stderr="The connector is already in manifest-only format.",
+        )
+      
+      # Detect sus python files in the connector source directory
+      connector_source_code_dir = self.context.connector.code_directory / self.context.connector.technical_name.replace("-", "_")
+      for file in connector_source_code_dir.iterdir():
+        if file.name not in VALID_FILES:
+          self.invalid_files.append(file.name)
+      if self.invalid_files:
+        return StepResult(
+          step=self,
+          status=StepStatus.SKIPPED,
+          stdout=f"The connector has unrecognized source files: {self.invalid_files}",
+        )
+      
+      # Detect connector class name to make sure it's inherited from source declarative manifest 
+      # and does not override `streams` method
+      connector_source_py = (connector_source_code_dir / "source.py").read_text()
+
+      if "YamlDeclarativeSource" not in connector_source_py:
+        return StepResult(
+          step=self,
+          status=StepStatus.SKIPPED,
+          stdout="The connector does not use the YamlDeclarativeSource class.",
+        )
+      
+      if "def streams" in connector_source_py:
+        return StepResult(
+          step=self,
+          status=StepStatus.SKIPPED,
+          stdout="The connector overrides the streams method.",
+        )
+
+      return StepResult(
+        step=self, 
+        status=StepStatus.SUCCESS, 
+        stdout=f"{self.context.connector.technical_name} is a valid candidate for migration.")
+
+class StripConnector(Step):
+    context: ConnectorContext
+
+    title = "Strip the connector to manifest-only."
+
+    async def _run(self) -> StepResult:
+
+      # 1. Move manifest.yaml to the root level of the directory
+      self.logger.info(f"Moving manifest to the root level of the directory")
+      connector_source_code_dir = self.context.connector.code_directory / self.context.connector.technical_name.replace("-", "_")
+
+      manifest_file = connector_source_code_dir / "manifest.yaml"
+      manifest_file = manifest_file.rename(self.context.connector.code_directory / "manifest.yaml")
+
+      if manifest_file not in self.context.connector.code_directory.iterdir():
+        return StepResult(
+          step=self,
+          status=StepStatus.FAILURE,
+          stdout="Failed to move manifest.yaml to the root level of the directory."
+        )
+      
+      # We don't want to delete the source_<name> folder
```

Comment:
We actually would very much love to delete the folder. We don't need it for sure. We should probably not delete `unit_tests/integration`, but then again we should check if it exists and NOT proceed.

Reply:
Gotcha! I removed the folder from the "don't delete" list and will work on adding the check for unit_tests/integration